src/utils.c

```diff
@@ -367,6 +367,31 @@ int map_intd(char *s, char **v, int dv) {
     return n;
 }
 
+int maps_intd(char *s, char **v, int dv) {
+    int i, n = dv;
+
+    if (s == NULL) {
+        LOG_AND_RETURN(dv, "maps_intd: s=>NULL, v=%p, %s %s", v,
+                       v ? v[0] : "NULL", v ? v[1] : "NULL");
+    }
+
+    s = strip(s);
+
+    if (!*s)
+        LOG_AND_RETURN(dv, "maps_intd: s is empty");
+
+    if (v == NULL) {
+        if (s[0] != '+' && s[0] != '-' && (s[0] < '0' || s[0] > '9'))
+            LOG_AND_RETURN(dv, "maps_intd: s not a number: %s, v=%p, %s %s", s,
+                           v, v ? v[0] : "NULL", v ? v[1] : "NULL");
+        return atoi(s);
+    }
+    for (i = 0; v[i]; i++)
+        if (strlen(s) <= strlen(v[i]) && !strncasecmp(s, v[i], strlen(v[i])))
```

Comment:
This is equivalent with strcasecmp

Reply:
> This is equivalent with strcasecmp

Are you sure?

Check these four cases:
1. Check "socket" with "socketworks".
2. Check "socketworks" with "socketworks".
3. Check "sock" with "socketworks".
4. Check "socketworks" with "socket".

If we don't do the check `(strlen(s) <= strlen(v[i])` the case 4 is incorrectly handed. And this is the case that this patch is targeting. I accept that a "substring" can be used in the parameter list; but this "substring" match requires to be exact in case of the target is smaller than the passed name. So in any case "socketworks" can be processed as a "socket" module name.

Reply:
To perform exact match (case 2), strcasecmp can be used and checked for 0

https://man7.org/linux/man-pages/man3/strcasecmp.3.html

 strcasecmp(const char *s1, const char *s2);
       
DESCRIPTION         top
       The strcasecmp() function performs a byte-by-byte comparison of
       the strings s1 and s2, ignoring the case of the characters.  It
       returns an integer less than, equal to, or greater than zero if
       s1 is found, respectively, to be less than, to match, or be
       greater than s2.

Reply:
Hi @catalinii ,

The code is updated and cleaned. I hope you agree with the final implementation.